src/nvim/os/env.c

```diff
@@ -451,18 +455,38 @@ static char *vim_version_dir(const char *vimdir)
   return NULL;
 }
 
-/// If the string between "p" and "pend" ends in "name/", return "pend" minus
-/// the length of "name/".  Otherwise return "pend".
-static char *remove_tail(char *p, char *pend, char *name)
+/// If `dirname + "/"` precedes tail in the path, return the pointer to
+/// `dirname + "/" + tail`.  Otherwise return `tail`.
+///
+/// Examples (path = /usr/local/share/nvim/runtime/doc/help.txt):
+///
+///   tail    = help.txt
+///   dirname = doc
+///   -> doc/help.txt
+///
+///   tail    = doc/help.txt
+///   dirname = runtime
+///   -> runtime/doc/help.txt
+///
+///   tail    = runtime/doc/help.txt
+///   dirname = vim74
+///   -> runtime/doc/help.txt
+///
+/// @param path    Path to a file
+/// @param tail    A suffix of the path
+/// @param dirname The immediate path fragment before the tail
+/// @return The new tail including dirname or just tail
+static char *remove_tail(char *path, char *tail, char *dirname)
```

Comment:
But the param name `tail` should at least be changed, because it's clearly in conflict with the function name `remove_tail`.